src/item.cpp

```diff
@@ -1870,9 +1871,11 @@ nc_color item::color_in_inventory() const
         switch( rating ) {
             case EDIBLE:
             case TOO_FULL:
-                if( is_going_bad() ) {
+                if( preserves ) {
+                    // Nothing, canned food won't rot
```

Comment:
I think it's acceptable for the color to not display if the container will prevent rot entirely. I've always disliked not being able to sort out my canned food from my genuine perishables at a glance.